solidity/nft_token.sol

```diff
@@ -0,0 +1,254 @@
+import "./library/spl_token.sol";
+import "./library/system_instruction.sol";
+
+struct Position {
+    address whirlpool;      // 32
+    address position_mint;  // 32
+    uint128 liquidity;      // 16
+    int32 tick_lower_index; // 4
+    int32 tick_upper_index; // 4
+}
+
+@program_id("DPvSokZKEAYqh6jjmGDUyd4L881uAWg4rFVeA6rec2fr")
+contract nft_token {
+    address public orca;
+    address public whirlpool;
+    address public pdaProgram;
+    address public mintErc20;
+    address public pdaERC20Account;
+    // TODO: Change for the latter one in production
+    uint64 public pdaHeader = 0x55bdfe33;//0xd0f7407ae48fbcaa;
+    // PDA seed
+    bytes public constant pdaProgramSeed = "pdaProgram";
+    // PDA bump
+    bytes1 public pdaBump;
```

Comment:
What is this?

Reply:
https://www.brianfriel.xyz/understanding-program-derived-addresses/
"fudge factor" (int8 = 255 .. 0) for "push" potencial PDA off the ed25519 elliptic curve
https://solanacookbook.com/core-concepts/pdas.html#generating-pdas
"When building with PDAs, it is common to store the bump seed in the account data itself. This allows developers to easily validate a PDA without having to pass in the bump as an instruction argument."
We simply follow examples and practices.